pkg/api/config.go

```diff
@@ -438,9 +438,27 @@ func validateLiteralTestStep(fieldRoot string, step LiteralTestStep, seen sets.S
 		ret = append(ret, fmt.Errorf("%s: `commands` is required", fieldRoot))
 	}
 	ret = append(ret, validateResourceRequirements(fieldRoot+".resources", step.Resources)...)
+	ret = append(ret, validateCredentials(fieldRoot, step.Credentials)...)
 	return
 }
 
+func validateCredentials(fieldRoot string, credentials []CredentialReference) []error {
```

Comment:
Should validate that `name` and `mountPath` are not empty, for good measure.

Any objections to leaving `namespace` empty to get it from `ci-operator`'s current namespace (asked a similar question below)?